lucene/core/src/java/org/apache/lucene/index/IndexWriter.java

```diff
@@ -3321,8 +3395,11 @@ private long prepareCommitInternal() throws IOException {
    * below.  We also ensure that we pull the merge readers while holding {@code IndexWriter}'s lock.  Otherwise
    * we could see concurrent deletions/updates applied that do not belong to the segment.
```

Comment:
This Javadoc will need to be updated to reflect the broader use of this method.

Also, is `preparePointInTimeMerge` (without the `on`) a better name?